meorg_client/client.py

```diff
@@ -216,6 +217,36 @@ def logout(self):
             self.headers.pop("X-User-Id", None)
             self.headers.pop("X-Auth-Token", None)
 
+    def upload_files_parallel(self, files: list, n: int = 2):
+        """Upload files in parallel.
+
+        Parameters
+        ----------
+        files : list
+            List of file paths.
+        n : int, optional
+            Number of threads to use, by default 2
+
+        Returns
+        -------
+        list
+            List of dicts or response objects from upload_files.
+        """
+
+        # Ensure the object is actually iterable
+        files = mu.ensure_list(files)
+
+        # Sequential case, single file provided
+        if len(files) == 1:
+            return self.upload_files(files)
```

Comment:
I think we can remove this check (since `Pool` with nprocesses=1 should do the same thing.

Reply:
I'm not sure of the underlying code in a pool.map, but I feel like it will still attempt to establish a pool and farm out the job to the processes, generating additional overhead. This is a simple catch-all where the user has requested parallel upload with a single file. The return type is also a single dict (in the successful case), whereas the multiple file option is a list.

Unless of course we default everything to be a list, even if that list is a single object long?

Eventually, I would like the default to be parallel uploads, but during testing it is helpful to maintain the different options.

I think I'll keep this for now, but raise an issue to change the defaults to return a list, even if it is a degenerate (single length) list. It makes things easier going forward.

Reply:
Raised https://github.com/CABLE-LSM/meorg_client/issues/47